include/kodlab_mjbots_sdk/lcm_subscriber.h

```diff
@@ -1,85 +1,102 @@
-// BSD 3-Clause License
-// Copyright (c) 2021 The Trustees of the University of Pennsylvania. All Rights Reserved
-// Authors:
-// Shane Rozen-Levy <srozen01@seas.upenn.edu>
+/**
+ * @file lcm_subscriber.h
+ * @author Ethan Musser (emusser@seas.upenn.edu)
+ * @author J. Diego Caporale (jdcap@seas.upenn.edu)
+ * @author Shane Rozen-Levy (srozen01@seas.upenn.edu)
+ * @brief Provides LCM subscriber object for subscribing to onw or multiple LCM
+ * channels.
+ * @date 7/7/22
+ *
+ * @copyright Copyright 2022 The Trustees of the University of Pennsylvania. All
+ * rights reserved. BSD 3-Clause License
+ *
+ */
 
 #pragma once
+
+#include <map>
+#include <mutex>
 #include <string>
 #include "lcm/lcm-cpp.hpp"
 #include "real_time_tools/thread.hpp"
 #include "kodlab_mjbots_sdk/abstract_realtime_object.h"
+#include "kodlab_mjbots_sdk/lcm_message_handler.h"
+
+namespace kodlab
+{
 
-namespace kodlab {
-/*!
- * @brief An template class for an lcm subscriber. Subscribes to just one msg of type msg_type.
- * @tparam MessageClass the lcm msg type
+/**
+ * @brief LCM subscriber capable of subscribing to multiple channels
+ * @todo Implement subscription removal (i.e., unsubscribing)
  */
-template<class MessageClass>
-class LcmSubscriber : public AbstractRealtimeObject {
- public:
+class LcmSubscriber : public AbstractRealtimeObject
```

Comment:
I think its important that we only ever run one of these to prevent concurrency issues.

Reply:
So right now we only want one on a single thread, partly because of how we are running the handleTimeout(1000) right?

Should we actually include a singleton design pattern to enforce this? something like the threadsafe version found here?
https://stackoverflow.com/questions/1008019/c-singleton-design-pattern

Reply:
no. no non ono no no no. Singletons are horrible. I don't know a single good c++ programmer who likes singeltons.

Reply:
hahahah 
Why? I have used them before (not in C++ i guess) . 
If the intent is only one should exist, why not enforce that?

Reply:
More specifically are they are annoying to implement or dangerous? Or one because the other?

Reply:
I don't have a great understanding of the hard reasons not to, but I know at least at BDI they are a constant pain to work around and maintain.

Reply:
Reading more about it, it seems that the Singleton Pattern implies some level of globality too, which is not what we want. 
I didn't realize the full implications of the pattern in C++. 
I don't think it makes sense in this use case. 

The comments should make it clear though that currently it would be bad to have multiple subscribers on a thread, or a subscriber and anything else really.

Reply:
I will update the comments to indicate that concurrency issues may arise if multiple `LcmSubscriber` objects exist simultaneously.

Reply:
Warning provided in 2bfd50c4afc4e757c1dab2850b8696d291b398ef.